v2/internal/controllers/crd_compute_vmss_20201201_test.go

```diff
@@ -247,10 +250,13 @@ func Test_Compute_VMSS_CRUD(t *testing.T) {
 	tc.Expect(found).To(BeTrue())
 
 	// Delete VMSS
+	// We do this in stages to avoid race conditions between owner/owned resources
+	// that can make tests non-deterministic
 	tc.DeleteResourceAndWait(vmss)
+	tc.DeleteResourceAndWait(rg)
```

Comment:
`rg` wasn't being deleted here before, so you don't actually need to delete it here at all? It'll get cleaned up automatically by the test GC process so you can just not delete the rg in the test (like before)

Reply:
I added this to be consistent with other tests that explicitly delete the resource group already.

Reply:
Today we're not all that consistent (some tests delete at the end, some don't. Some use defer, some don't). Im actually going to be removing the extra deletes in a future PR where I fix the parallel delete problem, which is why I mentioned it here (because these will end up removed in that PR). The test framework itself always issues these deletes when the test completes, so there's no need to do it explicitly here.

The intent is that the tests can just be focused on the actions applicable to the test at hand, and the ancillary stuff like creating a namespace, deleting RG when done, etc are taken care of for free by the framework.